From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 28 Apr 2022 19:51:06 -0700
Subject: [PATCH] (Paper-1.18.2) Use setMessage instead of arraycopy

setMessage affects messages, filteredMessage, and renderMessages, we were only doing an arraycopy for messages

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 02ad7f0b7a9428f9c4b4a5e8156551653533d7d8..cab8559dba305d0067f7f1e02be8044b824675d9 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -886,8 +886,9 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         SignBlockEntity sign = new SignBlockEntity(new BlockPos(loc.getBlockX(), loc.getBlockY(), loc.getBlockZ()), Blocks.OAK_SIGN.defaultBlockState());
         sign.setColor(net.minecraft.world.item.DyeColor.byId(dyeColor.getWoolData()));
         sign.setHasGlowingText(hasGlowingText);
-        System.arraycopy(components, 0, sign.messages, 0, sign.messages.length);
-
+        for (int i = 0; i < components.length; i++) {
+            sign.setMessage(i, components[i]);
+        }
         getHandle().connection.send(sign.getUpdatePacket());
     }
     // Paper end
